### ISSUE TITLE
fork a background update across *e2es on running / pending + waiting.

### ISSUE THREAD

**Issue by jayunit100** [kubernetes/kubernetes#12307]
@timothysc has just suggested this, he can add details on motivation for this  later.  in short, 
- currently: we have pretty running/pending graphs for e2es that use certain utility methods.
- but.. why restrict? after all theres no reason to only calculate "running/pending" during density tests, after all, it can be easily calculated for all tests, all the time, 
- assumption: probably isnt a huge drain on the api server if done at low resolution (i guess every 10 seconds or so) to get these stats during e2es.

Of course, this might be done on a  by namespace basis or cumulatively.   not sure if api yet supports cumulative stats.

tl;dr, this issue suggests to run a global e2e running/pending stats in a background thread, so that the entire span of e2e tests push out these stats to prometheus

**Comment by timothysc** [comment]
We can publish this data through the prom-gateway to record across * tests.  In general, having a set of metrics that we can gather and record across all the tests is useful.  Each test may wish to publish their own, but we can probably denote that as well. 

cc @wojtek-t

**Comment by mbforbes** [comment]
may be fun also as a side project for @kubernetes/goog-control-plane
